src/main/java/org/ow2/proactive/catalog/service/exception/CatalogObjectAlreadyExistingException.java

```diff
@@ -25,15 +25,26 @@
  */
 package org.ow2.proactive.catalog.service.exception;
 
-import java.io.IOException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
 
 
 /**
+ * This Exception is thrown when a POST request for object creation has been
+ * received but such a CatalogObject already exists (bucketName + ObjectName)
+ * The HTTP status is 409 (Conflict because of duplication attempt)
+ *
  * @author ActiveEon Team
  */
-public class DefaultRawCatalogObjectsFolderNotFoundException extends IOException {
+@ResponseStatus(value = HttpStatus.CONFLICT)
+public class CatalogObjectAlreadyExistingException extends DataIntegrityViolationException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This class has 8 parents which is greater than 5 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)